src/approaches/probe_approach.py

```diff
@@ -0,0 +1,201 @@
+
+Example command line:
+    export OPENAI_API_KEY=<your API key>
+    python src/main.py --approach probe --seed 0 \
+        --strips_learner oracle \
+        --env pddl_blocks_procedural_tasks \
+        --num_train_tasks 3 \
+        --num_test_tasks 1 \
+        --debug
+
+Easier setting:
+    python src/main.py --approach probe --seed 0 \
+        --strips_learner oracle \
+        --env pddl_easy_delivery_procedural_tasks \
+        --pddl_easy_delivery_procedural_train_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_train_max_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_max_extra_newspapers 0 \
+        --num_train_tasks 5 \
+        --num_test_tasks 10 \
+        --debug
+"""
+from __future__ import annotations
+
+import logging
+from typing import Collection, Dict, FrozenSet, Iterator, List, Optional, \
+    Sequence, Set, Tuple
+
+from predicators.src import utils
+from predicators.src.approaches.open_loop_llm_approach import \
+    OpenLoopLLMApproach
+from predicators.src.llm_interface import OpenAILLM
+from predicators.src.planning import task_plan_with_option_plan_constraint
+from predicators.src.settings import CFG
+from predicators.src.structs import Box, Dataset, GroundAtom, Object, \
+    ParameterizedOption, Predicate, State, Task, Type, _GroundNSRT, _Option
+from predicators.src.utils import apply_operator, \
+    create_task_planning_heuristic, get_applicable_operators, \
+    run_policy_guided_astar
+
+
+class Probe(OpenLoopLLMApproach):
+    """Probe definition."""
+
+    def __init__(self, initial_predicates: Set[Predicate],
+                 initial_options: Set[ParameterizedOption], types: Set[Type],
+                 action_space: Box, train_tasks: List[Task]) -> None:
+        super().__init__(initial_predicates, initial_options, types,
+                         action_space, train_tasks)
+        # Set up the LLM.
+        self._llm = OpenAILLM(CFG.open_loop_llm_model_name)
+        # Set after learning.
+        self._prompt_prefix = ""
+        self.last_char = ")"
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "probe"
+
+    def _predict(self, state: State, atoms: Set[GroundAtom],
+                 goal: Set[GroundAtom], memory: Dict) -> _GroundNSRT:
+        # If we already have an abstract plan, execute the next step.
```

Comment:
let's refactor this and the parent class so that we can remove the redundant lines involving the memory. so, make a new helper function that gets called inside of _predict, and move the non-shared logic into that helper function.